src/core/reference/include/openvino/reference/string_tensor_unpack.hpp

```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include "openvino/core/shape.hpp"
+
+namespace ov {
+namespace reference {
+void string_tensor_unpack(const std::string* data,
+                          int32_t* out_begins,
+                          int32_t* out_ends,
+                          uint8_t* out_symbols,
+                          const size_t element_count) {
+    size_t offset = 0;
+    for (size_t i = 0; i < element_count; ++i) {
+        out_begins[i] = offset;
+        out_symbols = std::copy(data[i].begin(), data[i].end(), out_symbols);
+        offset += data[i].length();
+        out_ends[i] = offset;
```

Comment:
I can see the type of `offset` variable has been updated since my last review, the `out_begins` and `out_ends` are pointers to `int32_t` (it's to align with existing ov tokenizers code as I remember) 
so the `offset` or string length value should be casted do the desired type, to resolve it (unfortunately with a little risk of overflow).

It leads to the following CI build issue:

> reference/string_tensor_unpack.hpp(17): warning C4267: '=': conversion from 'size_t' to 'int32_t', possible loss of data
reference/string_tensor_unpack.hpp(20): warning C4267: '=': conversion from 'size_t' to 'int32_t', possible loss of data

To be considered:
```suggestion
    int32_t offset = 0;
    for (size_t i = 0; i < element_count; ++i) {
        out_begins[i] = offset;
        out_symbols = std::copy(data[i].begin(), data[i].end(), out_symbols);
        offset += static_cast<int32_t>(data[i].length());
        out_ends[i] = offset;
```

Reply:
Changed to `static_cast`. Is this overflow risk something we're okay with or should I look for other solution?